lte/gateway/python/magma/subscriberdb/client.py

```diff
@@ -77,32 +78,47 @@ def __init__(
         self._grpc_client_manager = grpc_client_manager
 
     async def _run(self) -> None:
-        subscribers = await self._get_subscribers()
-        if subscribers is not None:
-            self._process_subscribers(subscribers)
+        subscribers, no_updates = await self._get_subscribers()
+        # No need to update local db if it's already up-to-date.
+        if subscribers is None or no_updates:
+            return
+        self._process_subscribers(subscribers)
 
     async def _get_subscribers(self) -> [SubscriberData]:
         subscriberdb_cloud_client = self._grpc_client_manager.get_client()
         subscribers = []
         req_page_token = ""  # noqa: S105
         res_page_token = "start_token"  # noqa: S105
+        cur_digest = SubscribersDigest(
+            md5_base64_digest=self._store.get_current_digest(),
+        )
```

Comment:
Side note: I hate how Python convention is to aggressively wrap at 80 chars. It makes the code so much harder to quickly parse / read

Nothing to change in your PR, just evangelizing my position 🙂 

Support for my position from e.g. Linus: http://lkml.iu.edu/hypermail/linux/kernel/2005.3/08168.html

Reply:
This email piece should definitely be circulated more!